ingestion/src/metadata/utils/aws_client.py

```diff
@@ -0,0 +1,68 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Optional, Any
+
+from boto3 import Session
+
+from metadata.config.common import ConfigModel
+
+
+class AWSClientConfigModel(ConfigModel):
+    """
+    AWSClientConfigModel holds all config parameters required to instantiate an AWSClient.
+    """
+
+    aws_access_key_id: Optional[str]
+    aws_secret_access_key: Optional[str]
+    aws_session_token: Optional[str]
+    endpoint_url: Optional[str]
+    region_name: Optional[str]
+
+
+class AWSClient(object):
```

Comment:
Introduced a generic AWSClient based on boto3 Session

Reply:
No need to directly pass `object`. By default all `class` extend `object`

Reply:
Thanks for the changes 🚀